service/matching/task_queue_partition_manager.go

```diff
@@ -358,6 +358,14 @@ func (pm *taskQueuePartitionManagerImpl) unloadPhysicalQueue(unloadedDbq physica
 		// this is the default queue, unload the whole partition if it is not healthy
 		if pm.defaultQueue == unloadedDbq {
 			pm.unloadFromEngine()
+
+			// have to iterate over versionedQueues and call updatePhysicalTaskQueueGauge
+			pm.versionedQueuesLock.Lock()
+			for _, vq := range pm.versionedQueues {
```

Comment:
Just came across the notion of using `defer` while locking variables, such as `pm.versionedQueuesLock`. Would this also be recommended here since we could technically crash from line 364?
@carlydf @ShahabT

Reply:
I'd definitely go with `Lock` and `defer Unlock` pattern if the lock was acquired at the first line of function. Now that it's in if branches, defer does not result in obvious code flow (although technically correct in this case because we immediately return after the if block). But no one knows who this if block will change in the future.

Reply:
I see - well, it won't matter (for now) since the latest changes don't require us iterating over `versionedQueues` for the time being (since metrics are now updated in Start/Stop and stop on versioned queues is called inside of `pm.unloadFromEngine`)